lib/shared/addon/components/cluster-driver/driver-okecapi/component.js

```diff
@@ -542,13 +558,25 @@ export default Component.extend(ClusterDriver, {
     if (!get(this, 'config.kubernetesVersion')) {
       errors.push('Kubernetes Version is required')
     }
-    if (!get(this, 'config.clusterCidr')) {
-      errors.push('Cluster CIDR is required')
+    if (get(this, 'vcnCreationMode') === 'Existing') {
+      if (!get(this, 'config.controlPlaneSubnet')) {
+        errors.push('Kubernetes API endpoint subnet is required')
+      }
+      if (!get(this, 'config.loadBalancerSubnet')) {
+        errors.push('Load balancer subnet is required')
+      }
+      if (!get(this, 'config.workerNodeSubnet')) {
+        errors.push('Worker node subnet is required')
+      }
     }
-    if (!get(this, 'config.podCidr')) {
-      errors.push('Pod CIDR is required')
+    if ((get(this, 'config.cniType') === 'OCI_VCN_IP_NATIVE') && (get(this, 'vcnCreationMode') === 'Existing')) {
```

Comment:
This is the case regardless of VCN Creation Mode

Reply:
But for form validation, doesn't adding the existing vcn mode check make more sense? We don't show the Pod CIDR Block box unless vcnCreationMode is 'Existing'. If it's quickCreate the user doesn't have to input the field

Reply:
yeah that's true, having the extra check doesn't hurt